File: Hanoi/Javascript/hanoi.js
function hanoi(disk_number, start, end, intermediate){
	if(disk_number != 0){
		hanoi(disk_number-1, start, intermediate, end);
		// printf("moving disk n°%d from %d to %d\n", disk_number, start, end);
		hanoi(disk_number-1, intermediate, end, start);
	}
}

var start = new Date().getTime();
hanoi(30, 0,2, 1);
var end = new Date().getTime();
var time = end - start;
console.log('Execution time: ' + time + " ms");

File: Hanoi/Ruby/Hanoi.rb
def move(disk_number, start, last, intermediate)
    if disk_number != 0
      move(disk_number-1, start, intermediate, last)
      #puts "moving disk #{disk_number} from #{start} to #{last}"
      move(disk_number-1, intermediate, last, start)
    end
  end

#puts "ARGV[0].to_i = #{ARGV[0].to_i}"
move(ARGV[0].to_i,0, 2, 1)

puts "-------------------"
puts "End of ruby program"
